exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/join/TestHashJoinOutcome.java

```diff
@@ -0,0 +1,204 @@
+      .addRow(123)
+      .build();
+    nonEmptyInputRowSetLeft = operatorFixture.rowSetBuilder(inputSchemaLeft)
+      .addRow(123)
+      .build();
+
+    // Prepare various (empty/non-empty) containers for each side of the join
+    emptyInputRowSetLeft = operatorFixture.rowSetBuilder(inputSchemaLeft).build();
+    emptyInputRowSetRight = operatorFixture.rowSetBuilder(inputSchemaRight).build();
+
+    inputContainerRight.add(emptyInputRowSetRight.container());
+    inputContainerRight.add(nonEmptyInputRowSetRight.container());
+
+    inputContainerLeft.add(emptyInputRowSetLeft.container());
+    inputContainerLeft.add(nonEmptyInputRowSetLeft.container());
+
+    final PhysicalOperator mockPopConfig = new MockStorePOP(null);
+    mockOpContext(mockPopConfig, 0, 0);
+  }
+
+  @After
+  public void afterTest() {
+    emptyInputRowSetRight.clear();
+    emptyInputRowSetLeft.clear();
+    nonEmptyInputRowSetRight.clear();
+    nonEmptyInputRowSetLeft.clear();
+    inputContainerRight.clear();
+    inputOutcomesRight.clear();
+    inputContainerLeft.clear();
+    inputOutcomesLeft.clear();
+  }
+
+  /**
+   *  Run the Hash Join where one side has an uninitialized container (the 2nd one)
+   * @param innerEmpty True if the right side is the uninitialized
+   * @param specialOutcome What outcome the uninitialized container has
+   * @param returnedOutcome what result outcome is expected
+   */
+  private void testHashJoinOutcomes(boolean innerEmpty, RecordBatch.IterOutcome specialOutcome,
+                                    RecordBatch.IterOutcome returnedOutcome) {
+
+    inputOutcomesLeft.add(RecordBatch.IterOutcome.OK_NEW_SCHEMA);
+    inputOutcomesLeft.add( innerEmpty ? RecordBatch.IterOutcome.OK : specialOutcome);
+
+    inputOutcomesRight.add(RecordBatch.IterOutcome.OK_NEW_SCHEMA);
+    inputOutcomesRight.add( innerEmpty ? specialOutcome : RecordBatch.IterOutcome.OK);
+
+    final MockRecordBatch mockInputBatchRight = new MockRecordBatch(operatorFixture.getFragmentContext(), opContext,
+      innerEmpty ? uninitialized2ndInputContainersRight : inputContainerRight,
+      inputOutcomesRight, batchSchemaRight);
+    final MockRecordBatch mockInputBatchLeft = new MockRecordBatch(operatorFixture.getFragmentContext(), opContext,
+      innerEmpty ? inputContainerLeft : uninitialized2ndInputContainersLeft,
+      inputOutcomesLeft, batchSchemaLeft);
+
+    List<JoinCondition> conditions = Lists.newArrayList();
+
+    conditions.add(new JoinCondition( SqlKind.EQUALS.toString(),
+      FieldReference.getWithQuotedRef("leftcol"),
+      FieldReference.getWithQuotedRef("rightcol")));
+
+    HashJoinPOP hjConf = new HashJoinPOP(null, null, conditions, JoinRelType.INNER);
+
+    HashJoinBatch hjBatch = new HashJoinBatch(hjConf,operatorFixture.getFragmentContext(), mockInputBatchLeft, mockInputBatchRight );
+
+    assertTrue(hjBatch.next() == returnedOutcome ); // verify returned outcome
```

Comment:
may be rename this to expectedOutcome instead of returnedOutcome ?

Reply:
Good idea. Also the new test file name was conceived in a rush, in case there's a better one. This is the first set of unit tests for the Hash-Join operator (vs. executing a whole join query). Maybe we could add more HJ unit tests in this file later.